scripts/s3/slurm/slurm.taskprolog

```diff
@@ -68,9 +68,10 @@
   # If Intel MPI is loaded and not included in the Singularity list, add its libraries
   if [[ -n ${I_MPI_ROOT} && ${SINGULARITY_CONTAINLIBS} != *${I_MPI_ROOT}* ]]; then
     I_MPI_LIBRARIES=$(find ${I_MPI_ROOT}/lib/{,mpi} -maxdepth 1 -type "f,l" | grep -vE "debug|release" | tr '\n' ',')
+    EFA_LIBRARIES=$(find /opt/amazon/efa/lib64 -maxdepth 1 -type "f,l" | tr '\n' ',')
     echo "export SINGULARITY_BIND=$(echo -n ${I_MPI_ROOT}/libfabric/lib,${I_MPI_ROOT}/etc,${SINGULARITY_BIND} | sed -r "s|(.*),$|\1|")"
-    echo "export SINGULARITY_CONTAINLIBS=$(echo -n ${I_MPI_LIBRARIES}${SINGULARITY_CONTAINLIBS} | sed -r "s|(.*),$|\1|")"
-    echo "export SINGULARITYENV_LD_PRELOAD=/.singularity.d/libs/libmpi.so"
+    echo "export SINGULARITY_CONTAINLIBS=$(echo -n ${I_MPI_LIBRARIES}${EFA_LIBRARIES}${SINGULARITY_CONTAINLIBS} | sed -r "s|(.*),$|\1|")"
+    echo "export SINGULARITYENV_LD_PRELOAD=/.singularity.d/libs/libmpi.so:/.singularity.d/libs/libfabric.so"
     # Check whether "EFA" is supported in this compute instance and force enable it if it is
     if [[ -n $(fi_info -p efa -t FI_EP_RDM 2>&1 | grep -e "provider: efa") ]]; then
         echo "export I_MPI_OFI_LIBRARY_INTERNAL=0"
```

Comment:
We are going to merge it "as is", no worries, but it would had made more sense for those changes to be inside the if-statement of line 75, right? Hehe.

Reply:
Ha, I tried that at first, prepending another time to `$SINGULARITY_CONTAINLIBS`, but then since it's `echo "export ..."` the variable gets expanded already without the result from the first time. 

Then I thought, I can change the construction and only `echo "export ..."` once at the end, but I couldn't be bothered. Those libraries are there regardless of which node, and it can run with that `libfabric` regardless of which node, the only thing is that there's no `efa` provider.

But I agree that it could be cleaned up!